src/safeds/data/tabular/transformation/_robust_scaler.py

```diff
@@ -0,0 +1,200 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from safeds._validation import _check_columns_exist
+from safeds._validation._check_columns_are_numeric import _check_columns_are_numeric
+from safeds.data.tabular.containers import Table
+from safeds.exceptions import TransformerNotFittedError
+
+from ._invertible_table_transformer import InvertibleTableTransformer
+
+if TYPE_CHECKING:
+    import polars as pl
+
+
+class RobustScaler(InvertibleTableTransformer):
+    """
+    The RobustScaler transforms column values to a range by removing the median and scaling to the interquartile range.
+
+    Currently for columns with high stability (IQR == 0) it will only substract the median and not scale to avoid dividing by zero.
+
+    Parameters
+    ----------
+    column_names:
+        The list of columns used to fit the transformer. If `None`, all numeric columns are used.
+    """
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Dunder methods
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def __init__(self, *, column_names: str | list[str] | None = None) -> None:
+        super().__init__(column_names)
+
+        # Internal state
+        self._data_median: pl.DataFrame | None = None
+        self._data_scale: pl.DataFrame | None = None
+
+    def __hash__(self) -> int:
+        # Leave out the internal state for faster hashing
+        return super().__hash__()
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Properties
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @property
+    def is_fitted(self) -> bool:
+        """Whether the transformer is fitted."""
+        return self._data_median is not None and self._data_scale is not None
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Learning and transformation
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def fit(self, table: Table) -> RobustScaler:
+        """
+        Learn a transformation for a set of columns in a table.
+
+        This transformer is not modified.
```

Comment:
```suggestion
        **Note:** This transformer is not modified.
```

For consistency with more "modern" implementations of transformers. We'll update the others later.